numpy/lib/_shape_base_impl.py

```diff
@@ -728,9 +728,7 @@ def dstack(tup):
 
 def _replace_zero_by_x_arrays(sub_arys):
     for i in range(len(sub_arys)):
-        if _nx.ndim(sub_arys[i]) == 0:
-            sub_arys[i] = _nx.empty(0, dtype=sub_arys[i].dtype)
-        elif _nx.sometrue(_nx.equal(_nx.shape(sub_arys[i]), 0)):
+        if _nx.ndim(sub_arys[i]) == 0 or _nx.sometrue(_nx.equal(_nx.shape(sub_arys[i]), 0)):
```

Comment:
I don't mind the combination, but this would read better if broken across two lines.

Reply:
I could do that, but if numpy starts using a formatter some day, the formatter won't split the line.